src/Microsoft.Diagnostics.Monitoring.WebApi/Operation/EgressOperation.cs

```diff
@@ -76,7 +74,22 @@ public void Validate(IServiceProvider serviceProvider)
         {
             serviceProvider
                 .GetRequiredService<IEgressService>()
-                .ValidateProvider(EgressProviderName);
+                .ValidateProviderExists(EgressProviderName);
+        }
+
+        public static async Task<ExecutionResult<EgressResult>> ValidateAsync(IServiceProvider serviceProvider, string endpointName, CancellationToken token)
```

Comment:
```suggestion
        public static async Task ValidateAsync(IServiceProvider serviceProvider, string endpointName, CancellationToken token)
```

Caller shouldn't have to inspect the result so don't provide it.